pkg/operator/ceph/object/bucket/rgw-handlers.go

```diff
@@ -101,6 +101,16 @@ func (p *Provisioner) deleteOBCResource(bucketName string) error {
 		} else if errors.Is(err, admin.ErrNoSuchBucket) {
```

Comment:
If ErrNoSuchKey is returned, we know the bucket doesn't exist, right? Maybe we don't need to call GetBucketInfo() to check again. But if there is another case `RemoveBucket()` might return ErrNoSuchKey then you can ignore my suggestion.
```suggestion
		} else if errors.Is(err, admin.ErrNoSuchBucket) || errors.Is(err, admin.ErrNoSuchKey) {
```

Reply:
Yes, there is another case. Ceph might return ErrNoSuchKey after confirming the existence of bucket.
If we regards ErrNoSuchKey as ErrNoSuchBucket, we can't distinguish the non-existence of the bucket and the failure of removing bucket.

https://github.com/ceph/ceph/blob/0ccd1f69be073987972f01e5e99fc1c19234034d/src/rgw/rgw_rest_bucket.cc#L234

Reply:
Ok thanks for the clarification